connector/connect/server/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -1068,6 +1068,10 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
   // Compares proto plan with LogicalPlan.
   private def comparePlans(connectPlan: proto.Relation, sparkPlan: LogicalPlan): Unit = {
     val connectAnalyzed = analyzePlan(transform(connectPlan))
-    comparePlans(connectAnalyzed, sparkPlan, false)
+    (connectAnalyzed, sparkPlan) match {
```

Comment:
In the current scenario, `connectAnalyzed` is transformed from `proto.Relation`. When it is `CollectMetrics`, the `dataframeId` is always 0.

But if the `sparkPlan` is `CollectMetrics`, its `dataframeId` value is determined by its corresponding DataFrame.

In the sbt test, `SparkConnectProtoSuite` is tested earlier, `sparkTestRelation` is the first created `DataFrame` with `id` as 0, thus the GA test didn't trigger the failure described in the pr.

When using Maven for testing, `SparkConnectProtoSuite` is tested later, `sparkTestRelation` is not the first created `DataFrame` with `id` not being 0, thus causing the test to fail.

Reply:
Thanks for the clarification!